### PR TITLE
test swift jobs with the queuedAt field intro

### PR DESCRIPTION
**Pull request by thoven87**
TODO Close after upstream merged

### CODE REVIEW

**Review by adam-fowler — CHANGES_REQUESTED** [review]
Can we add the new tests in swift-jobs here, so we can verify everything works with redis

**Comment by thoven87** [comment]
> Can we add the new tests in swift-jobs here, so we can verify everything works with redis

Will do!

File: Package.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@ let package = Package(
         .library(name: "JobsRedis", targets: ["JobsRedis"]),
     ],
     dependencies: [
-        .package(url: "https://github.com/hummingbird-project/swift-jobs.git", from: "1.0.0-beta.1"),
+        .package(url: "https://github.com/thoven87/swift-jobs.git", branch: "stevenson/queued-at"),
```
Comment (thoven87) [inline Package.swift:13]: TODO: Update after swift jobs [PR](https://github.com/hummingbird-project/swift-jobs/pull/11) is merged

**Comment by thoven87** [comment]
> Can we add the new tests in swift-jobs here, so we can verify everything works with redis

There's already a JobTest for Redis [here](https://github.com/hummingbird-project/swift-jobs-redis/blob/main/Tests/JobsRedisTests/RedisJobsTests.swift) Unlike the Postgres implementation, the queuedAt is not being persisted.

Does it make sense to stream the JobData `struct` introduce in the Postgres PR and save it as the buffer here?

**Review by adam-fowler — APPROVED** [review]

**Comment by adam-fowler** [comment]
> > Can we add the new tests in swift-jobs here, so we can verify everything works with redis
> 
> There's already a JobTest for Redis [here](https://github.com/hummingbird-project/swift-jobs-redis/blob/main/Tests/JobsRedisTests/RedisJobsTests.swift) Unlike the Postgres implementation, the queuedAt is not being persisted.
> 
> Does it make sense to stream the JobData `struct` introduce in the Postgres PR and save it as the buffer here?

Oh god this is quite awkward. Not sure how you should save the queuedAt time. Currently the redis implementation provides the equivalent of a set of key, value pairs. Where the key is the jobID and the value is the ByteBuffer of encoded data. Augmenting that with an additional value is going to be quite awkward or suboptimal

**Comment by adam-fowler** [comment]
One solution is to do this at the swift-jobs level and include it in the struct we encode. Then the Postgres and redis implementations don't need to know about it at all.

**Comment by thoven87** [comment]
> One solution is to do this at the swift-jobs level and include it in the struct we encode. Then the Postgres and redis implementations don't need to know about it at all.
> 
> EDIT: I meant to also add. I've looked at how we could do this, but it is a little messy and I want to clean up the surrounding code first.

Makes sense. Have you any idea when the clean up process will be ready?

**Comment by adam-fowler** [comment]
> Makes sense. Have you any idea when the clean up process will be ready?

I'll try to get something for tomorrow. Basically the issue is the encoding and decoding of jobs are done from different types and the types are spread across multiple files. A job to encode only needs the parameters and job id, a decoded job needs the parameters and job definition so it can run the job.

**Pull request closed without merging**